indico/modules/events/abstracts/models/abstracts.py

```diff
@@ -82,46 +89,63 @@ def __table_args__(cls):
             lazy=True
         )
     )
-    #: Data stored in abstract/contribution fields
-    field_values = db.relationship(
-        'AbstractFieldValue',
+    track = db.relationship(
+        'Track',
         lazy=True,
-        cascade='all, delete-orphan',
+        foreign_keys=[track_id],
         backref=db.backref(
-            'abstract',
-            lazy=True
+            'abstracts',
+            lazy='dynamic'
         )
     )
     type = db.relationship(
         'ContributionType',
         lazy=True,
         foreign_keys=[type_id],
         backref=db.backref(
-            'abstracts',
+            'proposed_abstracts',
             lazy=True
         )
     )
-    accepted_type = db.relationship(
+    final_type = db.relationship(
         'ContributionType',
         lazy=True,
-        foreign_keys=[accepted_type_id],
+        foreign_keys=[final_type_id],
+        backref=db.backref(
+            'completed_abstracts',
+            lazy=True
+        )
+    )
+    final_track = db.relationship(
+        'Track',
+        lazy=True,
+        foreign_keys=[final_track_id],
         backref=db.backref(
-            'accepted_as_abstracts',
+            'completed_abstracts',
             lazy=True
         )
     )
+    #: Data stored in abstract/contribution fields
+    field_values = db.relationship(
+        'AbstractFieldValue',
+        lazy=True,
+        cascade='all, delete-orphan',
+        backref=db.backref(
+            'abstract',
+            lazy=True
+        )
+    )
+
     # relationship backrefs:
     # - contribution (Contribution.abstract)
+    # - email_logs (AbstractEmailLog.abstract)
     # - judgments (Judgment.abstract)
```

Comment:
Yes, relationships between to-be-removed legacy models and new abstract models don't make much sense - we'll never use those together anyway